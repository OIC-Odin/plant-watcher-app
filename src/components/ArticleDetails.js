import React, { Component } from 'react';
import { View, Text, Image, StyleSheet, ScrollView } from 'react-native';
import { Rating } from 'react-native-elements';
import { Icon } from 'react-native-elements';

export default class ArticleDetails extends Component {
  convertTimestamp(timestamp) {
    const dt = new Date();
    const year = dt.getFullYear(timestamp);
    const month = dt.getMonth(timestamp);
    const date = dt.getDate(timestamp);

    return year + '/' + month + '/' + date;
  }

  editOrBookmark(editable) {
    if (editable === true) {
      return (
        <Text style={styles.edit}>編集</Text>
      );
    } else {
      return (
        <Icon type='font-awesome' name='bookmark-o' color='#D04255' size={30}/>
      );
    }
  }

  render() {
    const {
      body,
      editOrBookmarkBlock,
      block1,
      uploadImage,
      blockChild1,
      articleTitle,
      uploadDay,
      block2,
      bookmarked,
      article,
      review,
      border,
      rating
    } = styles;

    const {
      plantName,
      articleContents,
      createdAt,
      currentRating,
      image,
      editable,
    } = this.props;

    return (
      <ScrollView style={body}>
        <View style={editOrBookmarkBlock}>
          {this.editOrBookmark(editable)}
        </View>
        <View style={block1}>
          <Image
            source={image ? { uri: image } : require('../assets/images/image-not-found-small.png')}
            style={uploadImage}
          />
          <View style={blockChild1}>
            <Text style={articleTitle}>{plantName}</Text>
            <Text style={uploadDay}>{this.convertTimestamp(createdAt)}</Text>
          </View>
        </View>
        <View style={block2}>
          <Text style={bookmarked}>0ブックマーク</Text>
          <Rating
            type='star'
            readonly
            imageSize={18}
            startingValue={currentRating || 0}
          />
        </View>
        <View style={border}/>
        <View>
          <Text style={article}>{articleContents}</Text>
        </View>
        <Text style={review}>レビュー</Text>
        <View style={border}/>
        <Rating
          type='star'
          showRating
          startingValue={3}
          imageSize={30}
          onFinishRating={this.ratingCompleted}
          style={rating}
        />
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  body: {
    padding: 30,
  },
  editOrBookmarkBlock: {
    position: 'absolute',
    top: -10,
    right: 5
  },
  block1: {
    display: 'flex',
    flexDirection: 'row'
  },
  uploadImage: {
    width: 80,
    height: 80
  },
  blockChild1: {
    marginLeft: 18,
  },
  articleTitle: {
    paddingBottom: 5,
    fontSize: 21
  },
  uploadDay: {
    color: '#95989A',
    fontSize: 12
  },
  block2: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    marginTop: 15,
    marginBottom: 20
  },
  bookmarked: {
    marginRight: 18,
    fontSize: 12
  },
  article: {
    fontSize: 15,
    color: '#555555',
    letterSpacing: 1.5,
    lineHeight: 15 * 1.3,
  },
  review: {
    textAlign: 'center',
    marginTop: 40,
    marginBottom: 5,
    fontSize: 18
  },
  border: {
    marginBottom: 12,
    borderBottomColor: '#D1D1D6',
    borderBottomWidth: 1
  },
  rating: {
    marginBottom: 60
  },
  edit: {
    fontSize: 18,
    color: '#6BC6FD',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderStyle: 'solid',
    borderWidth: 1,
    borderColor: '#6BC6FD',
    borderRadius: 3
  }
});
