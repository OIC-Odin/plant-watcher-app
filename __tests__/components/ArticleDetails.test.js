import React from 'react';
import ArticleDetails from '../../src/components/ArticleDetails';
import renderer from 'react-test-renderer';

describe('記事詳細', () => {
  it('snapshot-記事が表示される', () => {
    const INITIAL_STATE = {
      articleDetails: {
        article: {
          plantName: 'ひまわり',
          articleContents: 'テスト',
          createdAt: 1510641275203,
          currentRating: 3.0
        },
      },
      editable: false,
      fetchArticleDetails: () => {},
    };
    const tree = renderer.create(<ArticleDetails {...INITIAL_STATE} />);
    expect(tree).toMatchSnapshot();
  });
});