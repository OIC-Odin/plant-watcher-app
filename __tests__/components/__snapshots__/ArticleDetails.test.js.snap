// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`記事詳細 snapshot-記事が表示される 1`] = `
<RCTScrollView
  style={
    Object {
      "padding": 30,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 5,
          "top": -10,
        }
      }
    >
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#D04255",
                "fontSize": 30,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 80,
            "width": 80,
          }
        }
      />
      <View
        style={
          Object {
            "marginLeft": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 21,
              "paddingBottom": 5,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#95989A",
              "fontSize": 12,
            }
          }
        >
          2018/0/7
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 20,
          "marginTop": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
            "marginRight": 18,
          }
        }
      >
        0ブックマーク
      </Text>
      <View
        pointerEvents="none"
        style={undefined}
      >
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "left": 0,
                "position": "absolute",
                "top": 0,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#f1c40f",
                  "height": 18,
                  "width": 0,
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "white",
                  "height": 18,
                  "width": 90,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </View>
          <View
            style={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 18,
                  "width": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#D1D1D6",
          "borderBottomWidth": 1,
          "marginBottom": 12,
        }
      }
    />
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#555555",
            "fontSize": 15,
            "letterSpacing": 1.5,
            "lineHeight": 19.5,
          }
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 40,
          "textAlign": "center",
        }
      }
    >
      レビュー
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#D1D1D6",
          "borderBottomWidth": 1,
          "marginBottom": 12,
        }
      }
    />
    <View
      pointerEvents="auto"
      style={
        Object {
          "marginBottom": 60,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#34495e",
                  "fontFamily": "Trebuchet MS",
                  "fontSize": 15,
                  "textAlign": "center",
                },
              ]
            }
          >
            Rating: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "fontFamily": "Trebuchet MS",
                    "fontSize": 30,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#f1c40f",
                  },
                ],
              ]
            }
          >
            3
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#34495e",
                  "fontFamily": "Trebuchet MS",
                  "fontSize": 18,
                  "textAlign": "center",
                },
              ]
            }
          >
            /
            5
          </Text>
        </View>
        <View />
      </View>
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "left": 0,
              "position": "absolute",
              "top": 0,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "#f1c40f",
                "height": 30,
                "width": 90,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "white",
                "height": 30,
                "width": 60,
              }
            }
          />
        </View>
        <View
          style={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          style={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          style={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          style={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
        <View
          style={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 30,
                "width": 30,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
